contributing.md

```diff
@@ -0,0 +1,369 @@
+
+# Build protocol buffer files (for development)
+make build_proto
+```
+
+## Contributing Guidelines
+
+### Code Style
+
+We use several tools to maintain code quality:
+
+- **Black** for code formatting
+- **isort** for import sorting
+- **flake8** for linting
+- **mypy** for type checking
+
+Run all checks before submitting:
+```bash
+# Format code
+make fmt
+
+# Check formatting and linting
+make check
+```
+
+### Commit Messages
+
+We follow the [Conventional Commits](https://www.conventionalcommits.org/) specification:
+
+```
+<type>[optional scope]: <description>
+
+[optional body]
+
+[optional footer(s)]
+```
+
+**Types:**
+- `feat`: A new feature
+- `fix`: A bug fix
+- `docs`: Documentation only changes
+- `style`: Changes that do not affect the meaning of the code
+- `refactor`: A code change that neither fixes a bug nor adds a feature
+- `test`: Adding missing tests or correcting existing tests
+- `chore`: Changes to the build process or auxiliary tools
+
+**Examples:**
+```
+feat(documentai): add signature detection support
+fix(cli): resolve authentication error on Windows
+docs(examples): add real estate analysis example
+test(functions): add unit tests for graph validation
+```
+
+### Branch Naming
+
+Use descriptive branch names with prefixes:
+- `feature/description` - For new features
+- `fix/description` - For bug fixes
+- `docs/description` - For documentation changes
+- `test/description` - For test improvements
+
+Examples:
+- `feature/signature-detection`
+- `fix/cli-auth-windows`
+- `docs/contributing-guide`
+
+## Pull Request Process
+
+### Before Creating a PR
+
+1. **Update your fork**
+   ```bash
+   git checkout main
+   git pull upstream main
+   git push origin main
+   ```
+
+2. **Create a feature branch**
+   ```bash
+   git checkout -b feature/your-feature-name
+   ```
+
+3. **Make your changes**
+   - Write clean, well-documented code
+   - Add tests for new functionality
+   - Update documentation as needed
+
+4. **Test your changes**
+   ```bash
+   make test 
+   make test_document_ai
+   make check
+   ```
+
+### Creating the PR
+
+1. **Push your branch**
+   ```bash
+   git push origin feature/your-feature-name
+   ```
+
+2. **Create the Pull Request**
+   - Use a clear, descriptive title
+   - Fill out the PR template completely
+   - Link any related issues
+   - Add screenshots or examples if applicable
+
+### PR Requirements
+
+- ✅ All tests pass (`make test`)
+- ✅ Code follows style guidelines (`make check`)
+- ✅ New functionality includes tests
+- ✅ Documentation is updated
+- ✅ No merge conflicts with main branch
+- ✅ PR description clearly explains the changes
+
+### Review Process
+
+1. **Automated checks** - CI/CD will run tests and checks
+2. **Code review** - A maintainer will review your code
+3. **Feedback** - Address any requested changes
+4. **Approval** - Once approved, a maintainer will merge your PR
+
+## Testing
+
+### Running Tests
+
+```bash
+# Run Workflow specific tests
+make test
```

Comment:
I think I saw this already

Reply:
This was more on writing them, I adjusted